suprb/optimizer/rule/ns/ns.py

```diff
@@ -122,13 +99,18 @@ def optimize(self, X: np.ndarray, y: np.ndarray, n_rules: int) -> list[Rule]:
             Return: A list of filtered Rules.
         """
         self._validate_params(n_rules)
-
         self.random_state_ = check_random_state(self.random_state)
+        self.novelty_calculation.archive.set_random_state(self.random_state_)
```

Comment:
l.102 is doing `check_random_state` which returns a `RandomState = Union[np.random.RandomState, np.random.Generator]`.
l.103 is taking that `RandomState` without doing `check_random_state` (as this would create another `RandomState`)

Since we already created one `RandomState` before, simply using that one should be enough. If we want to call `check_random_state` again, we can use `self.random_state` and initialize another `RandomState` within `Archive`